src/marks/lasso.js

```diff
@@ -0,0 +1,165 @@
+import {create, select, dispatch as dispatcher, line, pointer, polygonContains, curveNatural} from "d3";
+import {maybeTuple} from "../options.js";
+import {Mark} from "../plot.js";
+import {selection, selectionEquals} from "../selection.js";
+import {applyIndirectStyles} from "../style.js";
+
+const defaults = {
+  ariaLabel: "lasso",
+  fill: "#777",
+  fillOpacity: 0.3,
+  stroke: "#666",
+  strokeWidth: 2
+};
+
+export class Lasso extends Mark {
+  constructor(data, {x, y, ...options} = {}) {
+    super(
+      data,
+      [
+        {name: "x", value: x, scale: "x"},
+        {name: "y", value: y, scale: "y"}
+      ],
+      options,
+      defaults
+    );
+    this.activeElement = null;
+  }
+
+  // The lasso polygons follow the even-odd rule in css, matching the way
+  // they are computed by polygonContains.
+  render(index, scales, {x: X, y: Y}, dimensions) {
+    const margin = 5;
+    const {ariaLabel, ariaDescription, ariaHidden, fill, fillOpacity, stroke, strokeWidth} = this;
+    const {marginLeft, width, marginRight, marginTop, height, marginBottom} = dimensions;
+    if (!this.lasso) this.lasso = [];
```

Comment:
Am I missing something? I don’t see this.lasso being read anywhere. We only push onto it during render. So I’m guessing it could be removed.

More generally, I expect marks to be able to be used across plots. Like, you can create a mark instance, and then pass it to multiple plot instances. So we probably should avoid storing any interaction state on the mark instance itself. Possibly we’ll need Plot to provide a state store that is initialized when Plot.plot is called, so that a mark has a place to store interaction state.

Reply:
that was a bit of cruft, now removed; thanks!